nncf/dynamic_graph/transform_graph.py

```diff
@@ -34,24 +34,46 @@ def is_nncf_module(module):
     return False
 
 
-def replace_module_by_nncf_module(module: nn.Module):
+def replace_module_by_nncf_module(module: nn.Module, child_scope, eval_ops_exec_ctx_str=None):
+    replaced_module = module
     for nncf_module_type, module_type in NNCF_MODULES_DICT.items():
         if module.__class__.__name__ == module_type.__name__:
-            nncf_module = module
+            replaced_module = module
             if not module.__class__.__name__ == nncf_module_type.__name__:
-                nncf_module = nncf_module_type.from_module(module)
-            return nncf_module
+                replaced_module = nncf_module_type.from_module(module)
+
     from nncf.layers import UNWRAPPED_USER_MODULES
     for _, user_module_type in UNWRAPPED_USER_MODULES.registry_dict.items():
         if module.__class__ == user_module_type:
-            nncf_module = deepcopy(module)
-            return add_nncf_functionality_to_user_module(nncf_module)
-    return module
+            replaced_module = deepcopy(module)
+            replaced_module = add_nncf_functionality_to_user_module(replaced_module)
+
+    if module is not replaced_module:
+        if not module.weight.requires_grad:
+            nncf_logger.info("Ignored wrapping modules in scope: {} because "
+                             "the layer appears to be frozen (requires_grad=False)".format(child_scope))
```

Comment:
As stated in #201:
>That's why all frozen layers must be **explicitly** removed from the compression via ignored_scope attribute.
>If there are frozen layers not specified in ignored_scope, the error should happen with the message suggesting to >add frozen layers to ignored_scope or to make them unfrozen.

The logic I'm seeing, however, implicitly uses `requires_grad=False` to exclude modules from compression instead of checking whether the module is also within `ignored_scopes`. Has the strategy changed?

Reply:
not actual, as logic was changed